systemtest/src/test/java/io/strimzi/systemtest/security/SecurityST.java

```diff
@@ -1328,6 +1335,148 @@ void testOwnerReferenceOfCASecrets() {
+        X509Certificate userCert = getCertificateFromSecret(userSecret, "user.crt");
+        Date initialUserCertStartTime = userCert.getNotBefore();
+        Date initialUserCertEndTime = userCert.getNotAfter();
+
+        String dateKafkaUserCertCreation = userSecret.getMetadata().getManagedFields().get(0).getTime();
+        ZonedDateTime initialKafkaUserCertCreationDate = ZonedDateTime.parse(dateKafkaUserCertCreation);
+
+        LOGGER.info("Change of kafka validity and renewal days - reconciliation should start.");
+        CertificateAuthority newClientsCA = new CertificateAuthority();
+        newClientsCA.setRenewalDays(150);
+        newClientsCA.setValidityDays(200);
+        KafkaResource.replaceKafkaResource(clusterName, k -> k.getSpec().setClientsCa(newClientsCA));
+
+        // Wait for reconciliation and verify certs have been updated
+        Map<String, String> kafkaPods = StatefulSetUtils.ssSnapshot(KafkaResources.kafkaStatefulSetName(clusterName));
+        StatefulSetUtils.waitTillSsHasRolled(KafkaResources.kafkaStatefulSetName(clusterName), 3, kafkaPods);
+        KafkaUtils.waitForKafkaReady(clusterName);
+
+        // Read renewed secret/certs again
+        clientsCASecret = getClusterSecret("clients-ca-cert");
+        cacert = getCertificateFromSecret(clientsCASecret, "ca.crt");
+        Date changedCertStartTime = cacert.getNotBefore();
+        Date changedCertEndTime = cacert.getNotAfter();
+
+        userSecret = kubeClient().getSecret(username);
+        dateKafkaUserCertCreation = userSecret.getMetadata().getManagedFields().get(0).getTime();
+        ZonedDateTime changedKafkaUserCertCreationDate = ZonedDateTime.parse(dateKafkaUserCertCreation);
+
+        userCert = getCertificateFromSecret(userSecret, "user.crt");
+        Date changedUserCertStartTime = userCert.getNotBefore();
+        Date changedUserCertEndTime = userCert.getNotAfter();
+
+        LOGGER.info("Initial ClientsCA cert dates: " + initialCertStartTime + " --> " + initialCertEndTime);
+        LOGGER.info("Changed ClientsCA cert dates: " + changedCertStartTime + " --> " + changedCertEndTime);
+        LOGGER.info("Initial userCert dates: " + initialUserCertStartTime + " --> " + initialUserCertEndTime);
+        LOGGER.info("Changed userCert dates: " + changedUserCertStartTime + " --> " + changedUserCertEndTime);
+        LOGGER.info("KafkaUser cert creation dates: " + initialKafkaUserCertCreationDate + " --> " + changedKafkaUserCertCreationDate);
+
+        String msg = "Original Cert date end: '" + initialCertEndTime +
+                "' sooner than changed cert date '" + changedCertEndTime + "'";
+        assertThat(msg, initialCertEndTime.compareTo(changedCertEndTime) < 0);
+        assertThat("UserCert certificates were renewed", initialUserCertEndTime.compareTo(changedUserCertEndTime) < 0);
+        assertThat("KafkaUser 'user.crt' were renewed", initialKafkaUserCertCreationDate.compareTo(changedKafkaUserCertCreationDate) < 0);
```

Comment:
I expect that `certs were renewed` is correct behavior so the logic of the `reason` in assert is now wrong. Or I just misunderstand it?

Reply:
It compares dates. `OldCertDate < NewCertDate` (compareTo() has to be negative to make such boolean "true".

Reply:
so `assertThat("UserCert certificates were renewed", initialUserCertEndTime.compareTo(changedUserCertEndTime) < 0); ` fails if certs were renewed?

Reply:
Got it, negated 'reason' in fixup commit.